### ISSUE TITLE
C-j, C-k no longer working in hybrid style in helm

### ISSUE THREAD

**Issue by nixmaniack** [syl20bnr/spacemacs#5484]
With recent changes in develop, it looks like hybrid might be broken. I haven't tested extensively but this is the first instance which I have encountered after updating to 82fdd9a

**Comment by sooheon** [comment]
This is not a bug, but an intended [change](https://github.com/syl20bnr/spacemacs/commit/82fdd9a511f1f5cc4f371b91ce691308686f8ec6#diff-f81ff68d78d5c3218a2448dbbe9b57f0L396). For C-j/k to have vim-like movement, you may want to use the vim editing style. 

Rationale is [here](https://github.com/syl20bnr/spacemacs/issues/5237#issuecomment-189084814).

**Comment by nixmaniack** [comment]
@sooheon Thanks for referencing those points. But it looks like till those layer variables are added, it'd be considered a bug. 

> For C-j/k to have vim-like movement, you may want to use the vim editing style.

Nah! Hybrid is one of the primary reasons I like spacemacs! Maybe I was too habitual of using C-j/k in helm hence I thought of it as a bug. Time to retrain..

**Comment by syl20bnr** [comment]
Indeed this is a bug, I will look into hybrid mode during the week, I'll add the variables at this moment. I'm not sure yet what approach I will take with styles but it should be similar to layer variables.

**Comment by syl20bnr** [comment]
This is pushed to develop branch, example of configuration:

```elisp
dotspacemacs-editing-style '(hybrid :variables
                                    hybrid-mode-enable-evilified-state t
                                    hybrid-mode-enable-hjkl-bindings t
                                    hybrid-mode-default-state 'normal)
```

This will set the hybrid style the closest to vim style.

By default the variables are set to be the closest to the emacs style, ie:

```elisp
dotspacemacs-editing-style '(hybrid :variables
                                    hybrid-mode-enable-evilified-state nil
                                    hybrid-mode-enable-hjkl-bindings nil
                                    hybrid-mode-default-state 'hybrid)
```

**Comment by sooheon** [comment]
Thanks for this refactoring, it's helping spacemacs really live up to its goal of vim *and* emacs. IMHO, the default variable `hybrid-mode-default-state` should be 'normal. The docstring for `dotspacemacs-editing-style` states:

```
`hybrid' uses emacs key bindings for vim's insert mode, but otherwise leaves evil
unchanged.
```

In other words, hybrid mode should be a replacement of the insert state within evil-mode, but should not change how evil-mode starts up. Being in normal by default is one of the cornerstones of vi, after all. It makes sense that emacs style would start in emacs state, as there evil is seen as an addition.

**Comment by syl20bnr** [comment]
Good point, I'll revert it to normal.

**Comment by nixmaniack** [comment]
Thank you @syl20bnr! It's perfect.

**Comment by syl20bnr** [comment]
Done for the default value, it has been reverted to `normal`.

**Comment by aaronjensen** [comment]
@syl20bnr you may also consider setting `hybrid-mode-enable-evilified-state` to `t` by default as well. I opened this thinking it was a regression (which i guess it kind of was): https://github.com/syl20bnr/spacemacs/issues/5645

**Comment by syl20bnr** [comment]
@aaronjensen you are right, evilified state is not an insert state so it should be like Vim.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
